src/compiler/chainrules.jl

```diff
@@ -18,19 +17,68 @@ such that if a suitable rule is defined later, the generated function will recom
 """
 function has_chain_rrule(T)
   config_T, arg_Ts = Iterators.peel(T.parameters)
-  m_with_config = meta(Tuple{typeof(rrule), config_T, arg_Ts...})
-  if m_with_config.method === rrule_redispatcher_method
-    # it is being redispatched without config, so check it that hits the fallback
-    m_without_config = meta(Tuple{typeof(rrule), arg_Ts...})
-    if m_without_config.method === rrule_fallback_method
-      # no rrule exists, return instance for m_with_config as that will be invalidated 
-      # directly if configured rule added, or indirectly if unconfigured rule added
-      return false, m_with_config.instance
-    end
+  configured_rrule_m = meta(Tuple{typeof(rrule), config_T, arg_Ts...})
+  if _is_rrule_redispatcher(configured_rrule_m.method)
+    # The config is not being used:
+    # it is being redispatched without config, so we need the method it redispatches to
+    rrule_m = meta(Tuple{typeof(rrule), arg_Ts...})
+    # Thus any no_rrule that might apply must also not have a config because if there was a
+    # no_rrule with a config that applied then there would also be a rrule with config that applied
+    no_rrule_m = meta(Tuple{typeof(ChainRulesCore.no_rrule), arg_Ts...})
+  else
+    # Not being redispatched: it does have a config
+    rrule_m = configured_rrule_m
+     # Thus any no_rrule that might apply must also have a config because if it applied
+     # it will be identical, and if it doesn't we don't care what it is.
```

Comment:
```suggestion
    # Thus any no_rrule that might apply must also have a config because if it applied
    # it will be identical, and if it doesn't we don't care what it is.
```